### PR TITLE
VirtualWan: Use hubvnet CRUD operations for Hubvnet and hub cmdlets

### PR DESCRIPTION
**Pull request by shyamshd**
## Description

- Put/Delete hubvnet operations were using PUT on parent hub. Instead just use the newly exposed put/delete APIs on hubvnet connection.
- Use hubvnet connections crud operations in virtual hub cmdlets also. The virtual network connections here will be eventually deprecated. Add a warning to that effect.
- In, New hubvnet command, make enableInternetSecurity to be true by default unless specified otherwise. Since the parameter was previously a switch, add a deprecation warning for the old one and add a new flag.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by VeryEarly** [comment]
Hi @Shyamal-V 

Besides the review from @nagula-ritvika , it seems like there are build issue caused by SDK, please resolve inorder to pass CI checks

**Comment by shyamshd** [comment]
> Hi @Shyamal-V
> 
> Besides the review from @nagula-ritvika , it seems like there are build issue caused by SDK, please resolve inorder to pass CI checks

The build issues are because the SDK is not available yet. I built/tested it by creating a local SDK. The rest-api changes have been merged. I was told the new 2020-05-01 api version should be available this week. I uploaded it sooner to give the reviewers enough time.

**Comment by VeryEarly** [comment]
> > Hi @Shyamal-V
> > Besides the review from @nagula-ritvika , it seems like there are build issue caused by SDK, please resolve inorder to pass CI checks
> 
> The build issues are because the SDK is not available yet. I built/tested it by creating a local SDK. The rest-api changes have been merged. I was told the new 2020-05-01 api version should be available this week. I uploaded it sooner to give the reviewers enough time.

Please let me know once it was available, thanks.

**Comment by shyamshd** [comment]
@VeryEarly do you know when the new SDK would become available? Also, how do I trigger the pipeline to run again when it becomes available?

**Comment by VeryEarly** [comment]
> @VeryEarly do you know when the new SDK would become available? Also, how do I trigger the pipeline to run again when it becomes available?

SDK generation should be handled by service team.
You can do "/azp run" to restart the CI.

File: src/Network/Network/Cortex/VirtualHub/UpdateAzureRmVirtualHubCommand.cs [hunk 1]
```diff
@@ -77,6 +78,8 @@ public class UpdateAzureRmVirtualHubCommand : VirtualHubBaseCmdlet
             HelpMessage = "The address space string for this virtual hub.")]
         public string AddressPrefix { get; set; }
 
+        public const String ChangeDesc = "VirtualNetworkConnection parameter is deprecated. Use AzVirtualHubVnet commands";
```
Comment (nagula-ritvika) [inline src/Network/Network/Cortex/VirtualHub/UpdateAzureRmVirtualHubCommand.cs:81]: same : expand the full name of the cmdlet to use.

File: src/Network/Network/Cortex/VirtualHub/VirtualHubBaseCmdlet.cs [hunk 2]
```diff
@@ -94,6 +111,32 @@ public PSVirtualHub CreateOrUpdateVirtualHub(string resourceGroupName, string vi
...
 
+            if (psHubVnetConnections == null)
+            {
+                return hubToReturn;
+            }
+
+            // Crud operation on the vnet connections
+            hubToReturn.VirtualNetworkConnections = new List<PSHubVirtualNetworkConnection>();
+
+            foreach (var psConnection in psHubVnetConnections)
```
Comment (nagula-ritvika) [inline src/Network/Network/Cortex/VirtualHub/VirtualHubBaseCmdlet.cs:122]: 👍

Comment (shyamshd) [reply src/Network/Network/Cortex/VirtualHub/VirtualHubBaseCmdlet.cs:122]: Added change log. And changed message to *VirtualHubVnetConnection* to capture all below commands and their AzureRM counterparts-
Cmdlet          New-AzVirtualHubVnetConnection                     3.0.0      Az.Network
Cmdlet          Remove-AzVirtualHubVnetConnection                  3.0.0      Az.Network
Cmdlet          Update-AzVirtualHubVnetConnection                  3.0.0      Az.Network

**Comment by VeryEarly** [comment]
Hi @shyamshd ,

Please try pull latest changes from network-may branch, this should fix test case failures

**Comment by VeryEarly** [comment]
there are test cases need to be recorded, please check the test log

**Comment by shyamshd** [comment]
@VeryEarly I updated the test cases, but I am still running into an issue. It is unrelated to my change and in express route. Do we know if the test passes in the baseline without my changes?

**Comment by VeryEarly** [comment]
> @VeryEarly I updated the test cases, but I am still running into an issue. It is unrelated to my change and in express route. Do we know if the test passes in the baseline without my changes?
> 
> Also another question- Is there a way to run the pipeline with a different resource name and/or in a different region. I tried to debug the ER gateway failure & I suspect it might be to do with some zombie state related to this specific resource in the specific region.

This test case referenced New-AzVirtualHub which has API changed, please investigate and fix test case.

https://github.com/shyamshd/azure-powershell/blob/shdharma/deprecate-hubvnet-conn-from-hub/src/Network/Network.Test/ScenarioTests/VirtualHubRouteTableTests.ps1#L47

**Comment by VeryEarly** [comment]
close as merged in another PR #12171

**Pull request closed without merging**